### DISCUSSION TITLE
MessageLineLength size?

### DISCUSSION THREAD

**Discussion by Tuxified** [lintje/lintje#31] [Q&A] [answered]
Hi,

Before using `lintje` I already used a max line length of 80 for commit messages. Lintje "enforces" a line length of 72, I was wondering what the reasoning behind it is?

**Comment by tombruijn** [comment, marked as answer]
Hi @Tuxified,

Great question! I adopted that from Tim Pope's post [A Note About Git Commit Messages](https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html).

> Let’s start with a few of the reasons why wrapping your commit messages to 72 columns is a good thing.
> 
> - git log doesn’t do any special special wrapping of the commit messages. With the default pager of less -S, this means your paragraphs flow far off the edge of the screen, making them difficult to read. On an 80 column terminal, if we subtract 4 columns for the indent on the left and 4 more for symmetry on the right, we’re left with 72 columns.
> - git format-patch --stdout converts a series of commits to a series of emails, using the messages for the message body. Good email netiquette dictates we wrap our plain text emails such that there’s room for a few levels of nested reply indicators without overflow in an 80 column terminal. (The current rails.git workflow doesn’t include email, but who knows what the future will bring.)

I don't have very strong opinion about 80 character terminals or 72 character width specifically. In the future, I can potentially add a config option* to Lintje that you can choose between 72 or 80 characters, or it can starts printing hints (not errors) about line length at anything over 72 characters and only error about 80 charcters. In the end though, there needs to be some kind of reasonable maximum line length.

*: Lintje is supposed to be config free as much as possible though 😅

Let me know if you have any further thoughts about it 🙂
